processing/src/main/java/org/apache/druid/query/rowsandcols/ArrayListRowsAndColumns.java

```diff
@@ -153,11 +154,28 @@ public Column findColumn(String name)
       return new LimitedColumn(retVal, startOffset, endOffset);
     }
 
-    final Function<RowType, Object> adapterForValue = rowAdapter.columnFunction(name);
     final Optional<ColumnType> maybeColumnType = rowSignature.getColumnType(name);
     final ColumnType columnType = maybeColumnType.orElse(ColumnType.UNKNOWN_COMPLEX);
     final Comparator<Object> comparator = Comparator.nullsFirst(columnType.getStrategy());
 
+    final Function<RowType, Object> adapterForValue;
+    if (columnType.equals(ColumnType.STRING)) {
+      // special handling to reject MVDs
+      adapterForValue = f -> {
+        Object value = rowAdapter.columnFunction(name).apply(f);
+        if (value instanceof Object[] || value instanceof List) {
+          throw InvalidInput.exception(
+              "Encountered a multi value column [%s]. Window processing does not support MVDs. "
+              + "Try considering grouping on the column or use MV_TO_ARRAY()",
```

Comment:
Grammar isn't right here ("Try considering grouping"). Perhaps better to say: "Consider grouping on the column or using MV_TO_ARRAY." or even "Consider using UNNEST or MV_TO_ARRAY."

Reply:
Thanks for the review! will update it :)